### PR TITLE
Fix options for threshold computation in SpectrumObservation, fix tests

### PR DESCRIPTION
**Pull request by lmohrmann**
This PR introduces more sane defaults for the method options in `SpectrumObservation.compute_energy_threshold`, in particular it allows to set a method to `None` to not apply a threshold at the lower or upper end.

Also, fixing some tests that use this method.

### CODE REVIEW

**Comment by cdeil** [comment]
IMO the two indented if to handle the "None" case isn't so nice.
How about always passing a string, and calling it "none" or "off" for that case?
(uniform API this way is usually also easier to hook up to higher-level scripts that e.g. get the selection from a config file)

@joleroi - Please review.

File: gammapy/spectrum/tests/test_extract.py [hunk 1]
```diff
@@ -105,8 +105,8 @@ def test_sherpa(self, tmpdir, extraction):
         desired = extraction.observations[0].aeff.data.data.value
         assert_allclose(actual, desired)
 
-    def test_define_energy_threshold(self, extraction):
+    def test_compute_energy_threshold(self, extraction):
         # TODO: Find better API for this
```
Comment (joleroi) [inline gammapy/spectrum/tests/test_extract.py]: I think this TODO is resolved now, isn't it?

**Comment by lmohrmann** [comment]
@cdeil I like your suggestion, just implemented this.

File: gammapy/spectrum/tests/test_extract.py (hunk 1 above)
Comment (lmohrmann) [reply gammapy/spectrum/tests/test_extract.py]: I guess? Do you want me to remove it?

Comment (joleroi) [reply gammapy/spectrum/tests/test_extract.py]: Yes, please :wink:

**Comment by cdeil** [comment]
@lmohrmann - Thanks!

**Pull request merged**